src/main/java/bio/terra/service/snapshot/SnapshotService.java

```diff
@@ -359,6 +361,30 @@ public UUID getFirstSourceDatasetIdFromSnapshotId(
         .orElseThrow(() -> new DatasetNotFoundException("Source dataset for snapshot not found"));
   }
 
+  public SnapshotPreviewModel retrievePreview(
```

Comment:
Not a blocker, but it feels like we're growing "SnapshotService" too much using a generic object with too many responsibilities.